platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/events/DetailedConsensusEvent.java

```diff
@@ -25,54 +27,86 @@
  * information.
  */
 public class DetailedConsensusEvent extends AbstractSerializableHashable implements SelfSerializable, RunningHashable {
+    /** Value used to indicate that it is undefined*/
+    public static final long UNDEFINED = -1;
 
     public static final long CLASS_ID = 0xe250a9fbdcc4b1baL;
     public static final int CLASS_VERSION = 1;
 
     /** the pre-consensus event */
     private GossipEvent gossipEvent;
-    /** Consensus data calculated for an event */
-    private ConsensusData consensusData;
     /** the running hash of this event */
     private final RunningHash runningHash = new RunningHash();
+    /** the round in which this event received a consensus order and timestamp */
+    private long roundReceived = UNDEFINED;
+    /** true if this event is the last in consensus order of all those with the same received round */
+    private boolean lastInRoundReceived = false;
 
     /**
      * Creates an empty instance
      */
     public DetailedConsensusEvent() {}
 
+    /**
+     * Create a new instance.
+     *
+     * @param event the event to copy the data from
+     */
+    public DetailedConsensusEvent(final EventImpl event) {
+        this.gossipEvent = event.getBaseEvent();
+        this.roundReceived = event.getRoundReceived();
+        this.lastInRoundReceived = event.isLastInRoundReceived();
+    }
+
     /**
      * Create a new instance with the provided data.
      *
-     * @param gossipEvent   the pre-consensus event
-     * @param consensusData the consensus data for this event
+     * @param gossipEvent         the pre-consensus event
+     * @param roundReceived       the round in which this event received a consensus order and timestamp
+     * @param lastInRoundReceived true if this event is the last in consensus order of all those with the same received
+     *                            round
      */
-    public DetailedConsensusEvent(final GossipEvent gossipEvent, final ConsensusData consensusData) {
+    public DetailedConsensusEvent(
+            final GossipEvent gossipEvent, final long roundReceived, final boolean lastInRoundReceived) {
         this.gossipEvent = gossipEvent;
-        this.consensusData = consensusData;
+        this.roundReceived = roundReceived;
+        this.lastInRoundReceived = lastInRoundReceived;
     }
 
     public static void serialize(
-            final SerializableDataOutputStream out, final GossipEvent gossipEvent, final ConsensusData consensusData)
+            final SerializableDataOutputStream out,
+            final GossipEvent gossipEvent,
+            final long roundReceived,
+            final boolean lastInRoundReceived)
             throws IOException {
         gossipEvent.serialize(out);
-        out.writeSerializable(consensusData, false);
+
+        out.writeInt(ConsensusData.CLASS_VERSION);
+        out.writeLong(UNDEFINED); // ConsensusData.generation
```

Comment:
Those comments are not helpfull. Maybe you can add 3 more words to understand the meaning.

Reply:
i added a comment with an explenation